core/chain-configs/src/client_config.rs

```diff
@@ -69,6 +73,80 @@ impl GCConfig {
     }
 }
 
+fn default_num_concurrent_requests() -> u64 {
+    DEFAULT_STATE_SYNC_NUM_CONCURRENT_REQUESTS_EXTERNAL
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct ExternalStorageConfig {
+    /// Location of state parts.
+    pub location: ExternalStorageLocation,
+    /// When fetching state parts from external storage, throttle fetch requests
+    /// to this many concurrent requests per shard.
+    #[serde(default = "default_num_concurrent_requests")]
+    pub num_concurrent_requests: u64,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub enum ExternalStorageLocation {
+    S3 {
+        /// Location of state dumps on S3.
+        bucket: String,
+        /// Data may only be available in certain locations.
+        region: String,
+    },
+    Filesystem {
+        root_dir: PathBuf,
+    },
+}
+
+/// Configures how to dump state to external storage.
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug)]
+pub struct DumpConfig {
+    /// Specifies where to write the obtained state parts.
+    pub location: ExternalStorageLocation,
```

Comment:
can DumpConfig.location be different than SyncConfig::ExternalStorage's location?

Reply:
Yes